src/plugins/formulas/formulas.js

```diff
@@ -63,368 +62,294 @@ class Formulas extends BasePlugin {
+    this.addHook('modifySourceData', (...args) => this.onModifySourceData(...args));
+    this.addHook('modifyData', (...args) => this.onModifyData(...args));
+    this.addHook('beforeRemoveCol', (...args) => this.onBeforeRemoveCol(...args));
+    this.addHook('beforeRemoveRow', (...args) => this.onBeforeRemoveRow(...args));
+    this.addHook('beforeCreateCol', (...args) => this.onBeforeCreateCol(...args));
+    this.addHook('beforeCreateRow', (...args) => this.onBeforeCreateRow(...args));
     this.addHook('afterRemoveCol', (...args) => this.onAfterRemoveCol(...args));
     this.addHook('afterRemoveRow', (...args) => this.onAfterRemoveRow(...args));
-    this.addHook('afterSetDataAtCell', (...args) => this.onAfterSetDataAtCell(...args));
-    this.addHook('afterSetDataAtRowProp', (...args) => this.onAfterSetDataAtCell(...args));
+    this.addHook('afterCreateCol', (...args) => this.onAfterCreateCol(...args));
+    this.addHook('afterCreateRow', (...args) => this.onAfterCreateRow(...args));
     this.addHook('beforeColumnSort', (...args) => this.onBeforeColumnSort(...args));
-    this.addHook('beforeCreateCol', (...args) => this.onBeforeCreateCol(...args));
-    this.addHook('beforeCreateRow', (...args) => this.onBeforeCreateRow(...args));
-    this.addHook('beforeRemoveCol', (...args) => this.onBeforeRemoveCol(...args));
-    this.addHook('beforeRemoveRow', (...args) => this.onBeforeRemoveRow(...args));
-    this.addHook('beforeValidate', (...args) => this.onBeforeValidate(...args));
-    this.addHook('beforeValueRender', (...args) => this.onBeforeValueRender(...args));
-    this.addHook('modifyData', (...args) => this.onModifyData(...args));
+    this.addHook('afterColumnSort', (...args) => this.onAfterColumnSort(...args));
 
-    this.sheet.addLocalHook('afterRecalculate', (...args) => this.onSheetAfterRecalculate(...args));
+    // TODO: check if these hooks from the previous version of the plugin are still relevant
+    // this.addHook('afterSetDataAtCell', (...args) => this.onAfterSetDataAtCell(...args));
+    // this.addHook('afterSetDataAtRowProp', (...args) => this.onAfterSetDataAtCell(...args));
+    // this.addHook('beforeColumnSort', (...args) => this.onBeforeColumnSort(...args));
+    // this.addHook('beforeCreateCol', (...args) => this.onBeforeCreateCol(...args));
+    // this.addHook('beforeCreateRow', (...args) => this.onBeforeCreateRow(...args));
+    // this.addHook('beforeRemoveCol', (...args) => this.onBeforeRemoveCol(...args));
+    // this.addHook('beforeRemoveRow', (...args) => this.onBeforeRemoveRow(...args));
+    // this.addHook('beforeValidate', (...args) => this.onBeforeValidate(...args));
+    // this.addHook('beforeValueRender', (...args) => this.onBeforeValueRender(...args));
+
+    this.settings = this.hot.getSettings().formulas;
+
+    this.sheetName = this.hyperformula.addSheet(this.settings.sheetName || void 0);
+
+    this.sheetId = this.hyperformula.getSheetId(this.sheetName);
```

Comment:
I would keep just one. Prefarbly `sheetId`, and make the other one a getter:

```ts
get sheetName() {
  return this.hyperformula.getSheetName(this.sheetId);
}
```

When we add the ability to rename the sheet we will have to update this in multiple places. With the getter HyperFormula will be a single source of truth.